scripts/dev/check_constexpr.py

```diff
@@ -428,42 +463,37 @@ def constexpr_check(filepath: Path) -> list[LogMessage]:
         if line == "":
             continue
 
-        # skip comment lines
-        if line[0:2] == "//":
+        # skip comment lines (full-line C++ style)
+        if line.startswith("//"):
             continue
 
-        # strip trailing comments
+        # strip trailing comments (simple heuristic)
         if "//" in line:
-            tokens = line.split("//")
+            tokens = line.split("//", 1)
             line = tokens[0].strip()
 
-        re_match_1 = re.match(CONST_NUM_PATTERN, line)
-        re_match_2 = re.match(ARRAY_CONST_PATTERN, line)
-        re_match_3 = re.match(ARRAY_US_CONST_PATTERN, line)
-        if (re_match_1 or re_match_2 or re_match_3) and bracket_count == 0:
+        # remove string literals so we don't match inside them
+        line_no_strings = strip_string_literals(line)
+
+        # evaluate patterns on code-without-strings
+        re_match_1 = re.match(CONST_NUM_PATTERN, line_no_strings)
+        re_match_2 = re.match(ARRAY_CONST_PATTERN, line_no_strings)
+        re_match_3 = re.match(ARRAY_US_CONST_PATTERN, line_no_strings)
 
+        if (re_match_1 or re_match_2 or re_match_3) and bracket_count == 0:
             errors.append(
                 ErrorMessage(
                     tool="check_constexpr",
                     filepath=filepath,
                     line_number=idx,
-                    line=line,
+                    line=line,  # report the non-comment original line
                     message="Use 'constexpr' instead of 'const' for variable declaration",
                 )
             )
 
-        # count brackets in line
-        bracket_count += line.count("(")
-        bracket_count -= line.count(")")
-
-    if bracket_count != 0:
-        errors.append(
-            InfoMessage(
-                tool="check_constexpr",
-                filepath=filepath,
-                message=f"{bracket_count} unmatched parentheses",
-            )
-        )
```

Comment:
This was just counting parentheses in the files, and if they were unbalanced, throwing the message which appeared on PRs. I just removed it for now, since it doesn't appear to be that useful.